src/main/java/org/dependencytrack/policy/VersionPolicyEvaluator.java

```diff
@@ -50,25 +53,71 @@ public List<PolicyConditionViolation> evaluate(final Policy policy, final Compon
             return violations;
         }
 
+        
         final var componentVersion = new ComponentVersion(component.getVersion());
+        final var componentPurl = component.getPurl();
+        final var componentPurlType = componentPurl != null ? componentPurl.getType() : null;
+        Version componentVersionObj = null;
+
+        if (componentPurl != null && componentPurlType != null) {
+            componentVersionObj = VersionFactory.forScheme(componentPurlType, component.getVersion());
+        }
 
         for (final PolicyCondition condition : super.extractSupportedConditions(policy)) {
-            LOGGER.debug("Evaluating component (" + component.getUuid() + ") against policy condition (" + condition.getUuid() + ")");
+            LOGGER.debug("Evaluating component (" + component.getUuid() + ") against policy condition ("
+                    + condition.getUuid() + ")");
 
-            final var conditionVersion = new ComponentVersion(condition.getValue());
-            if (conditionVersion.getVersionParts().isEmpty()) {
-                LOGGER.warn("Unable to parse version (" + condition.getValue() + " provided by condition");
-                continue;
-            }
+            if (componentVersionObj == null) {
+                final var conditionVersion = new ComponentVersion(condition.getValue());
+                if (conditionVersion.getVersionParts().isEmpty()) {
+                    LOGGER.warn("Unable to parse version (" + condition.getValue() + " provided by condition");
+                    continue;
+                }
 
-            if (matches(componentVersion, conditionVersion, condition.getOperator())) {
-                violations.add(new PolicyConditionViolation(condition, component));
+                if (matches(componentVersion, conditionVersion, condition.getOperator())) {
+                    violations.add(new PolicyConditionViolation(condition, component));
+                }
+            } else {
+                final var conditionVersionObj = VersionFactory.forScheme(componentPurlType, condition.getValue());
+                if (conditionVersionObj == null){
+                    LOGGER.warn("Unable to parse version (" + condition.getValue() + " provided by condition");
+                    continue;
+                }
```

Comment:
Failures to construct a version will throw [`InvalidVersionException`](https://github.com/nscuro/versatile/blob/main/versatile-spi/src/main/java/io/github/nscuro/versatile/spi/InvalidVersionException.java)s, not return `null`.